lib/ts/recipe/recipeModule.ts

```diff
@@ -73,38 +73,32 @@ export default abstract class RecipeModule {
     };
 
     redirect = async (
-        context: { action: string },
+        context: { action: string; redirectToPath?: string },
         history?: History<LocationState>,
-        title?: string,
-        shouldReload = false
+        queryParams?: Record<string, string>
     ): Promise<void> => {
         const redirectUrl = await this.getRedirectionURL(context);
-
-        // If shouldReload, reload.
-        if (shouldReload === true) {
-            return await redirectToWithReload(redirectUrl);
-        }
         try {
             new URL(redirectUrl);
-            // Otherwise, If full URL, use redirectToWithReload
-            return await redirectToWithReload(redirectUrl);
+            // If full URL, use redirectToWithReload
+            return await redirectToWithReload(redirectUrl, queryParams);
         } catch (e) {
             // Otherwise, redirect in app.
-            return await redirectToInApp(redirectUrl, title, history);
+            return await redirectToInApp(redirectUrl, history, queryParams);
         }
     };
 
-    getRedirectionURL = async (context: { action: string }): Promise<string> => {
-        let redirectUrl;
+    getRedirectionURL = async (context: { action: string; redirectToPath?: string }): Promise<string> => {
         // If getRedirectionURL provided by user.
         const getRedirectionURL = this.hooks.getRedirectionURL;
```

Comment:
I prefer if this is normalised in the utils just like all other input configs. So in utils, we check if the user has provided this function, if not, we set the config to have the default function. This way, when using, we can directly call the function via config without worrying about undefined checks and without having this extra function here.